add advanced slack custom message use case (#1262)
checkly/docs.checklyhq.com: site/content/docs/integrations/slack.md

This approach is ideal when the native Slack integration does not meet your formatting or dynamic content needs.

### Setup Overview

You will be using a Webhook alert channel with a Slack Incoming WebHook URL and a custom payload template. This setup allows you to:

- Control Slack message formatting using Slack's [Block Kit](https://api.slack.com/block-kit)
- Link directly to a runbook or documentation for the failing check

### Steps

1. **Create a Slack Webhook**
   - [Create an Incoming WebHook integration](https://my.slack.com/services/new/incoming-webhook/) in Slack by selecting a default channel for your alerts. You’ll see a WebHook URL generated by Slack. Copy it.
   - Choose the bot name and icon for your alerts.

2. **Create a Webhook Alert Channel in Checkly**
   - Go to [Alert Settings > Add More Channels > Webhook](https://app.checklyhq.com/alerts/settings/channels/new/webhook)
   - Fill in the following:
     - **Name**: e.g. `Custom Slack Alerts`
     - **URL**: Paste the Slack Webhook URL
     - **Method**: `POST`
     - **Send Alerts for**: Enable `Failure`, `Recovery`, and `Degraded`

>      - **SSL Expiry**: Optional, not supported in this template

Reviewer: `Failure`, `Recovery`, and `Degraded` formatting implies we're quoting the options word-for-word, change to something like:

Example Line 63: `**Notification events**: Enable recovery, degraded, and failure. SSL expiry optional, but not supported in this template.`

Nit: if you merge line 64 with line 63 like the example above, it'll better match our UI format

>      - **Template**: See below

Reviewer: Nit wording: "Body" matches UI better, so something like:

`**Body**: see template below`

From the recovered Slack message, you can see the timestamp as well as a link to the check itself.

![A slack message showing a recovered alert](/docs/images/alerting/slack-recovered-check.png)

## Custom Slack Webhook Integration

For users who need more control over the Slack alert format and content, you can use a **Webhook Alert Channel** in Checkly to send fully customized messages to Slack.

This approach is ideal when the native Slack integration does not meet your formatting or dynamic content needs.

### Setup Overview

You will be using a Webhook alert channel with a Slack Incoming WebHook URL and a custom payload template. This setup allows you to:

- Control Slack message formatting using Slack's [Block Kit](https://api.slack.com/block-kit)

>   - Error messages (on failure)

Author: ```suggestion
```

From the recovered Slack message, you can see the timestamp as well as a link to the check itself.

![A slack message showing a recovered alert](/docs/images/alerting/slack-recovered-check.png)

## Custom Slack Webhook Integration

For users who need more control over the Slack alert format and content, you can use a **Webhook Alert Channel** in Checkly to send fully customized messages to Slack.

This approach is ideal when the native Slack integration does not meet your formatting or dynamic content needs.

### Setup Overview

You will be using a Webhook alert channel with a Slack Incoming WebHook URL and a custom payload template. This setup allows you to:

- Control Slack message formatting using Slack's [Block Kit](https://api.slack.com/block-kit)

>   - Links to runbooks or external OTel providers

Author: ```suggestion
```

From the recovered Slack message, you can see the timestamp as well as a link to the check itself.

![A slack message showing a recovered alert](/docs/images/alerting/slack-recovered-check.png)

## Custom Slack Webhook Integration

For users who need more control over the Slack alert format and content, you can use a **Webhook Alert Channel** in Checkly to send fully customized messages to Slack.

This approach is ideal when the native Slack integration does not meet your formatting or dynamic content needs.

### Setup Overview

You will be using a Webhook alert channel with a Slack Incoming WebHook URL and a custom payload template. This setup allows you to:

- Control Slack message formatting using Slack's [Block Kit](https://api.slack.com/block-kit)

> - Show detailed check information such as:

Author: ```suggestion
```

From the recovered Slack message, you can see the timestamp as well as a link to the check itself.

![A slack message showing a recovered alert](/docs/images/alerting/slack-recovered-check.png)

## Custom Slack Webhook Integration

For users who need more control over the Slack alert format and content, you can use a **Webhook Alert Channel** in Checkly to send fully customized messages to Slack.

This approach is ideal when the native Slack integration does not meet your formatting or dynamic content needs.

### Setup Overview

You will be using a Webhook alert channel with a Slack Incoming WebHook URL and a custom payload template. This setup allows you to:

- Control Slack message formatting using Slack's [Block Kit](https://api.slack.com/block-kit)
- Show detailed check information such as:

>   - Response time

Author: ```suggestion
  - Links to runbooks or external OTel providers
```

  - Error messages (on failure)

>   - Recovery or degraded state

Author: ```suggestion
```

From the recovered Slack message, you can see the timestamp as well as a link to the check itself.

![A slack message showing a recovered alert](/docs/images/alerting/slack-recovered-check.png)

## Custom Slack Webhook Integration

For users who need more control over the Slack alert format and content, you can use a **Webhook Alert Channel** in Checkly to send fully customized messages to Slack.

This approach is ideal when the native Slack integration does not meet your formatting or dynamic content needs.

### Setup Overview

You will be using a Webhook alert channel with a Slack Incoming WebHook URL and a custom payload template. This setup allows you to:

- Control Slack message formatting using Slack's [Block Kit](https://api.slack.com/block-kit)
- Show detailed check information such as:

>   - Run location and start time

Author: ```suggestion
```

From the recovered Slack message, you can see the timestamp as well as a link to the check itself.

![A slack message showing a recovered alert](/docs/images/alerting/slack-recovered-check.png)

> ## Custom Slack Webhook Integration (Advanced)

Author: ```suggestion
## Custom Slack Webhook Integration
```

> For advanced users who need more control over the Slack alert format and content, you can use a **Webhook Alert Channel** in Checkly to send fully customized messages to Slack.

Author: ```suggestion
For users who need more control over the Slack alert format and content, you can use a **Webhook Alert Channel** in Checkly to send fully customized messages to Slack.
```

This approach is ideal when the native Slack integration does not meet your formatting or dynamic content needs.

### Setup Overview

> You will be using a `WebhookAlertChannel` construct with a Slack Incoming WebHook URL and a custom payload template. This setup allows you to:

Author: ```suggestion
You will be using a Webhook alert channel with a Slack Incoming WebHook URL and a custom payload template. This setup allows you to:
```

- Control Slack message formatting using Slack's [Block Kit](https://api.slack.com/block-kit)
- Show detailed check information such as:

>   - Check name and ID

Author: ```suggestion
```

  - Run location and start time
  - Response time
  - Error messages (on failure)
  - Recovery or degraded state
- Link directly to a runbook or documentation for the failing check

> ### Setting It Up

Author: ```suggestion
### Steps
```

1. **Create a Slack Webhook**

>    - Follow Step 1 from the main section above to generate a Webhook URL in Slack.

Author: ```suggestion
   - [Create an Incoming WebHook integration](https://my.slack.com/services/new/incoming-webhook/) in Slack by selecting a default channel for your alerts. You’ll see a WebHook URL generated by Slack. Copy it.
```

   - Choose the bot name and icon for your alerts.

2. **Create a Webhook Alert Channel in Checkly**

>    - Go to **Alert Settings > Add More Channels > Webhook**

Author: ```suggestion
   - Go to [Alert Settings > Add More Channels > Webhook](https://app.checklyhq.com/alerts/settings/channels/new/webhook)
```

   - Fill in the following:
     - **Name**: e.g. `Custom Slack Alerts`
     - **URL**: Paste the Slack Webhook URL
     - **Method**: `POST`
     - **Send Alerts for**: Enable `Failure`, `Recovery`, and `Degraded`
     - **SSL Expiry**: Optional, not supported in this template
     - **Template**: See below

3. **Customize the Template**

The following template uses conditional logic (`{{#if}}`) to change the message depending on the alert type:

> ```

Author: ```suggestion
``` {title="Body"}
```

              {{else if (eq CHECK_NAME "Check B")}}
              "url": "https://example.com/check-b",
              {{else}}
              "url": "https://example.com/default",
              {{/if}}
              "action_id": "open_link_button"
            },
            {
              "type": "button",
              "text": {
                "type": "plain_text",
                "text": "OTel Provider",
                "emoji": true
              },
              "url": "https://example.com",
              "action_id": "open_link_button_needs_to_be_unique"
            }
          ]
        },
        {
          "type": "context",
          "elements": [
            {
              "type": "plain_text",
              "text": "Tags: {{TAGS}} | UUID: {{CHECK_RESULT_ID}}",
              "emoji": false
            }
          ]
        }
      ]
    }
  ]
}
```

>[!NOTE] 
> Be sure to update or remove the placeholder button links to Runbooks, OTel provider and anything else that is not needed for your use case.

>[!NOTE] 
> Be sure each check mapped in the runbook logic has a corresponding URL. If a match isn’t found, fallback to a default documentation page.

> ### Testing the Webhook limitations

Author: ```suggestion
### Testing the Webhook Limitations
```

From the recovered Slack message, you can see the timestamp as well as a link to the check itself.

![A slack message showing a recovered alert](/docs/images/alerting/slack-recovered-check.png)

> ---

Author: ```suggestion
```
